### PR TITLE
[WFCORE-4901] TCCL not set to application classloader when MBean Noti…

### PR DESCRIPTION
**Pull request by moulalis**
JIRA issue: https://issues.redhat.com/browse/WFCORE-4901

Old PR : https://github.com/wildfly/wildfly-core/pull/4126

### CODE REVIEW

**Comment by moulalis** [comment]
Hi @bstansberry ,
This is new PR for https://github.com/wildfly/wildfly-core/pull/4126 which I've closed.
Please review it.
Thanks.

File: jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java [hunk 1]
```diff
@@ -105,6 +112,33 @@ public void testReservedDomainMBeanRegistrationFails() throws Exception {
         reservedDomainTest("jboss.as.expr:bean=test-null", NAME);
     }
 
+    @Test
+    public void testTccl() throws Exception {
```
Comment (bstansberry) [inline jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java:116]: Minor: can you call this "testNotificationTccl"?  Or something like that.

File: jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java [hunk 2]
```diff
@@ -165,6 +199,10 @@ public void preDeregister() throws Exception {
         public void postDeregister() {
         }
 
+        @Override
+        public void proceed() {
+            sendNotification(new Notification("testtccl", 1, 1));
```
Comment (bstansberry) [inline jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java:204]: If you change the return type of this method to ClassLoader, then you can add this:

```
return WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
```

And then the test can validate that it returns newOutTccl. That both serves as a sanity check that newOutTccl was in place before the notification handling, and confirms that it got restored.

Comment (moulalis) [reply jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java:204]: @bstansberry  I've Renamed it to testNotificationTccl. Initailly I did the same as returntype ClassLoader, But I thought this is not invoking notification was just a test,  so I wrote it inside notification, In that way it actually invoked notification and test the same. Is that okay to have this or still you want me to change?

File: jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java (hunk 1 above)
Comment (moulalis) [reply jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java]: @bstansberry  I've changed it to testNotificationTccl

File: jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java (hunk 2 above)
Comment (moulalis) [reply jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java:204]: @bstansberry  Initailly I did the same as returntype ClassLoader, But I thought this is not invoking notification was just a test, so I wrote it inside notification, In that way it actually invoked notification and test the same. Is that okay to have this or still you want me to change?

Comment (bstansberry) [reply jmx/src/test/java/org/jboss/as/jmx/PluggableMBeanServerTestCase.java:204]: What your NotificationListener is doing is fine -- it checks that the expected TCCL is in place when it is invoked. I was suggesting an additional check that the expected TCCL (newOutTccl) is in place after the notification call is done.

But this is ok as is.

**Comment by jmesnil** [comment]
@moulalis thanks

**Pull request merged**